modules/node_modules/@ncigdc/components/DownloadTableToTsvButton.js

```diff
@@ -0,0 +1,64 @@
+// @flow
+import React from 'react';
+import { map, reduce } from 'lodash';
+
+import saveFile from '@ncigdc/utils/filesaver';
+import { mapStringArrayToTsvString } from '@ncigdc/utils/toTsvString';
+import Button from '@ncigdc/uikit/Button';
+import { visualizingButton } from '@ncigdc/theme/mixins';
+import { Tooltip } from '@ncigdc/uikit/Tooltip';
+
+type TProps = {
+  selector: string,
+  filename: string,
+  style: Object,
+};
+
+const getSingleHeader = (headThs: Array<NodeList>) => (
+  reduce(headThs[0], (acc, th) => (th.rowSpan === 2 ? [...acc, th] : [...acc, ...map(headThs[1], t => t)]), [])
+);
+
+const DownloadTableToTsvButton = ({
+  filename,
+  selector,
+  style = {},
+}: TProps) => (
+  <Tooltip Component={<span>Export current table view as TSV</span>} >
+    <Button
+      style={{ ...visualizingButton, ...style }}
+      onClick={() => {
+        const tableEl = document.querySelector(selector);
+        const headTrs = tableEl.querySelector('thead').querySelectorAll('tr');
+        const headThs = map(headTrs, h => h.querySelectorAll('th'));
+        const thEls = headThs.length === 2 ? getSingleHeader(headThs) : tableEl.querySelectorAll('th');
+        const thText = map(thEls, el => el.innerText).map(t => t.replace('\n', ' '));
+        const trs = tableEl.querySelector('tbody').querySelectorAll('tr');
+        const tdText = map(trs, t => {
+          const tds = t.querySelectorAll('td');
+          return reduce(tds, (acc, td) => {
+            const markedForTsv = td.querySelector('.for-tsv-export');
+            const exportText = markedForTsv ? markedForTsv.innerText : td.innerText;
+            const joinedText = exportText.trim().split('\n').join(',');
```

Comment:
Instead of requiring no `&nbsp;` could we just add something like `.replace(/\s*/g, ' ')` here?